src/Evolution/DgSubcell/Tags/Jacobians.hpp

```diff
@@ -76,33 +117,25 @@ struct InverseJacobianLogicalToInertialCompute
           std::string,
           std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>&
           functions_of_time,
-      std::optional<::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
-                                      Frame::Grid>>
+      const ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                              Frame::Grid>
           inverse_jacobian_logical_to_grid) {
-    if (inverse_jacobian_logical_to_grid.has_value()) {
-      // Assign
-      *inverse_jacobian_logical_to_inertial = std::optional<::InverseJacobian<
-          DataVector, Dim, Frame::ElementLogical, Frame::Inertial>>(
-          ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
-                            Frame::Inertial>(grid_coords.get(0).size()));
-      // Get grid to inertial inverse jacobian
-      const auto& inverse_jacobian_grid_to_inertial =
-          grid_to_inertial_map.inv_jacobian(grid_coords, time,
-                                            functions_of_time);
-      for (size_t i = 0; i < Dim; i++) {
-        for (size_t j = 0; j < Dim; j++) {
-          auto& inv_jacobian_component =
-              inverse_jacobian_logical_to_inertial->value().get(i, j);
-          inv_jacobian_component = 0.;
-          for (size_t k = 0; k < Dim; k++) {
-            inv_jacobian_component +=
-                inverse_jacobian_logical_to_grid.value().get(i, k) *
-                inverse_jacobian_grid_to_inertial.get(k, j);
-          }
```

Comment:
I would also do:
```cpp
if (grid_to_inertial_map.is_identity()) {
  // Optimize time-independent case by only setting pointers.
  if (UNLIKELY(not inverse_jacobian_logical_to_inertial->has_value())) {
    *inverse_jacobian_logical_to_inertial =
            ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
                              Frame::Inertial>(get<0>(grid_coords).size());
  }
  for (size_t storage_index = 0; storage_index < inverse_jacobian_logical_to_grid.size(); ++storage_index) {
    make_const_view(make_not_null(&std::as_const(inverse_jacobian_logical_to_inertial->value()[storage_index])),
       inverse_jacobian_logical_to_grid[storage_index], 0, inverse_jacobian_logical_to_grid[storage_index].size());
  }
} else {
  // All the code you currently have
}
```
This means that when the maps are time-dependent we don't compute anything, so the tag will be cheap to evaluate